src/Http/Controllers/V1/Admin/CMS/PageController.php

```diff
@@ -80,4 +81,24 @@ public function update(Request $request, $id)
             'message' => __('rest-api::app.common-response.success.update', ['name' => 'Page']),
         ]);
     }
+
+    /**
+     * To mass remove the cms page.
+     *
+     * @param  \Webkul\Core\Http\Requests\MassDestroyRequest  $request
+     * @return \Illuminate\Http\Response
+     */
+
+    public function massDestroyResources(MassDestroyRequest $request)
+    {
+        foreach ($request->indexes as $id) {
+            $this->getRepositoryInstance()->findOrFail($id);
+
+            $this->getRepositoryInstance()->delete($id);
+        }
```

Comment:
Earlier we are using the for each loop and this is having the drawback of multiple queries. For improvement purposes, I think we can make it one query also by using whereIn()->delete().